service/history/historyEngine.go

```diff
@@ -1929,6 +1930,7 @@ func (e *historyEngineImpl) SignalWithStartWorkflowExecution(
 			)
 		}
 		policy := workflow.WorkflowIdReusePolicyAllowDuplicate
```

Comment:
Previously `policy` was reference and allowed `nil`. So default value was `nil` and here it was changed to `WorkflowIdReusePolicyAllowDuplicate`. Now it never can be `nil` and if not passed it is set to 0, which is `WorkflowIdReusePolicyAllowDuplicateFailedOnly`. I have to fix tests to explicitly pass `WorkflowIdReusePolicyAllowDuplicate` but this is good example when we need `0` to be some `Unknown` value.